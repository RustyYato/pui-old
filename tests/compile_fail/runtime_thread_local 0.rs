pui::make_counter_tl! {
    type Foo = u8;
}

fn main() {
    fn assert_send_sync<T: Send + Sync>(_: T) {}
    assert_send_sync(Foo::new());
}
