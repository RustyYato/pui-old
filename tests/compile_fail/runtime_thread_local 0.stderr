error[E0277]: `*mut ()` cannot be sent between threads safely
 --> $DIR/runtime_thread_local 0.rs:7:5
  |
6 |     fn assert_send_sync<T: Send + Sync>(_: T) {}
  |                            ---- required by this bound in `main::assert_send_sync`
7 |     assert_send_sync(Foo::new());
  |     ^^^^^^^^^^^^^^^^ `*mut ()` cannot be sent between threads safely
  |
  = help: within `pui::runtime::Runtime<Foo>`, the trait `std::marker::Send` is not implemented for `*mut ()`
  = note: required because it appears within the type `pui::ThreadLocal`
  = note: required because it appears within the type `std::marker::PhantomData<pui::ThreadLocal>`
  = note: required because it appears within the type `Foo`
  = note: required because it appears within the type `pui::runtime::Runtime<Foo>`

error[E0277]: `*mut ()` cannot be shared between threads safely
 --> $DIR/runtime_thread_local 0.rs:7:5
  |
6 |     fn assert_send_sync<T: Send + Sync>(_: T) {}
  |                                   ---- required by this bound in `main::assert_send_sync`
7 |     assert_send_sync(Foo::new());
  |     ^^^^^^^^^^^^^^^^ `*mut ()` cannot be shared between threads safely
  |
  = help: within `pui::runtime::Runtime<Foo>`, the trait `std::marker::Sync` is not implemented for `*mut ()`
  = note: required because it appears within the type `pui::ThreadLocal`
  = note: required because it appears within the type `std::marker::PhantomData<pui::ThreadLocal>`
  = note: required because it appears within the type `Foo`
  = note: required because it appears within the type `pui::runtime::Runtime<Foo>`
